src/bluejay_schema_analyzer.rs

```
@@ -0,0 +1,171 @@
+use crate::scale_limits_analyzer::ScaleLimitsAnalyzer;
+use anyhow::{anyhow, Result};
+use bluejay_parser::{
+    ast::{
+        definition::{DefinitionDocument, SchemaDefinition},
+        executable::ExecutableDocument,
+        Parse,
+    },
+    Error,
+};
+use serde_json::to_string as to_json_string;
+
+pub struct BluejaySchemaAnalyzer;
+
+impl BluejaySchemaAnalyzer {
+    pub fn analyze_schema_definition(
+        schema_string: &str,
+        query: &str,
+        input: &serde_json::Value,
+    ) -> Result<f64> {
+        let document_definition = DefinitionDocument::parse(schema_string)
+            .map_err(|errors| anyhow!(Error::format_errors(schema_string, errors)))?;
+
+        let schema_definition = SchemaDefinition::try_from(&document_definition)
+            .map_err(|errors| anyhow!(Error::format_errors(schema_string, errors)))?;
+
+        let executable_document = ExecutableDocument::parse(query)
+            .map_err(|errors| anyhow!(Error::format_errors(query, errors)))?;
+
+        let cache =
+            bluejay_validator::executable::Cache::new(&executable_document, &schema_definition);
+
+        let input_str = to_json_string(input).unwrap_or_else(|_| "<invalid JSON>".to_string());
```

Instead of passing along `<invalid JSON>`, should we just return an error here?